### DISCUSSION TITLE
Modelling microwave heating

### DISCUSSION THREAD

**Discussion by AxelNil** [NanoComp/meep#2023] [Q&A] [answered]
Hi! We're a group of undergraduate CS students at Uppsala University currently working on a project where we want to use MEEP to simulate microwave heating of different materials. Is it possible to simulate the heating directly with MEEP or will we need to use auxiliary software to simulate the heat propagation? In that case, could you point us in the right direction?

Thanks in advance

**Comment by oskooi** [comment]
While it is not possible to  directly simulate thermal emission from a black body in Meep, it is possible to use [Kirchhoff's law](https://en.wikipedia.org/wiki/Kirchhoff%27s_law_of_thermal_radiation) which states that emissivity E(ω) is equivalent to absorptivity A(ω) at thermal equilibrium to then compute the absorptivity A(ω) of a *lossy* material system in Meep. The thermal emission spectrum of the object is just the product of the black body spectrum and its emissivity. For an example of this type of calculation, see: http://www.simpetus.com/projects.html#meep_thermal_radiation.

**Comment by oskooi** [comment, marked as answer]
Also, it is possible to compute a map of the absorbed power density of a device as demonstrated in: https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#absorbed-power-density-map-of-a-lossy-cylinder. This provides information on the local absorption of incident photon radiation. To simulate the phonon heat transfer *within* the device, you will need to use a different software package since Meep only involves Maxwell's equations.

**Comment by stevengj** [comment]
Right.  Basically, for heating you want to compute the absorbed power density, then use that as a source term in a [heat equation](https://en.wikipedia.org/wiki/Heat_equation) (with appropriate boundary conditions to capture the heat outflow) to compute the resulting temperature.
